base/shell/explorer/traydeskbtn.cpp

```diff
@@ -0,0 +1,443 @@
+/*
+ * PROJECT:     ReactOS Explorer
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Show Desktop tray button implementation
+ * COPYRIGHT:   Copyright 2006-2007 Thomas Weidenmueller <w3seek@reactos.org>
+ *              Copyright 2018-2022 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ *              Copyright 2023 Ethan Rodensky <splitwirez@gmail.com>
+ */
+
+#include "traydeskbtn.h"
+#include <limits.h>
+
+#define IDI_SHELL32_DESKTOP 35
+#define IDI_IMAGERES_DESKTOP 110
+
+#define SHOW_DESKTOP_TIMER_ID 999
+#define SHOW_DESKTOP_TIMER_INTERVAL 200
+
+CTrayShowDesktopButton::CTrayShowDesktopButton() :
+    m_nClickedTime(0),
+    m_inset({2, 2}),
+    m_icon(NULL),
+    m_highContrastMode(FALSE),
+    m_drawWithDedicatedBackground(FALSE),
+    m_bHovering(FALSE),
+    m_bPressed(FALSE),
+    IsHorizontal(FALSE)
+{
+}
+
+BOOL
+CTrayShowDesktopButton::GetTaskbar(OUT HWND* taskbarWnd)
+{
+    *taskbarWnd = NULL;
+    auto parent = GetParent();
+    if (!parent)
+        return FALSE;
+
+    parent = parent.GetParent();
+    if (!parent)
+        return FALSE;
+
+    HWND wnd = parent.m_hWnd;
+    *taskbarWnd = wnd;
+    return wnd != NULL;
+}
+
+INT CTrayShowDesktopButton::WidthOrHeight() const
+{
+    if (IsThemeActive() && !m_highContrastMode)
+    {
+        if (m_drawWithDedicatedBackground)
+        {
+            if (GetSystemMetrics(SM_TABLETPC))
+            {
+                //TODO: DPI scaling - return logical-to-physical conversion of 24, not fixed value
+                return 24;
+            }
+            else
+                return 15;
+        }
+        else
+        {
+            return max(16 + (IsHorizontal
+                ? (m_ContentMargins.cxLeftWidth + m_ContentMargins.cxRightWidth)
+                : (m_ContentMargins.cyTopHeight + m_ContentMargins.cyBottomHeight)
+            ), 18) + 6;
+        }
+    }
+    else
+    {
+        return max(
+            2 * GetSystemMetrics(SM_CXBORDER) + GetSystemMetrics(SM_CXSMICON),
+            2 * GetSystemMetrics(SM_CYBORDER) + GetSystemMetrics(SM_CYSMICON)
+        );
+    }
+}
+
+HRESULT CTrayShowDesktopButton::DoCreate(HWND hwndParent)
+{
+    DWORD style = WS_CHILD | WS_VISIBLE | WS_CLIPSIBLINGS | BS_DEFPUSHBUTTON;
+    Create(hwndParent, NULL, NULL, style);
+
+    if (!m_hWnd)
+        return E_FAIL;
+
+    bool bIconRetrievalFailed = ExtractIconExW(L"imageres.dll", -IDI_IMAGERES_DESKTOP, NULL, &m_icon, 1) == UINT_MAX;
+    if (bIconRetrievalFailed || !m_icon)
+        ExtractIconExW(L"shell32.dll", -IDI_SHELL32_DESKTOP, NULL, &m_icon, 1);
+    
+    m_szIcon.cx = 16;
+    m_szIcon.cy = 16;
```

Comment:
Not using `GetSystemMetrics(SM_CXSMICON)`, `GetSystemMetrics(SM_CYSMICON)` ?

Reply:
Why didn't I do that? Am I stupid?

Reply:
@HBelusca If you look a few lines below, you'll see that I use the dimensions of the icon resource - the 16 is only for if the icon resource couldn't be obtained, or if determining its dimensions fails (how???)

Refer to https://github.com/reactos/reactos/pull/5744#discussion_r1603588548 for context - this may now be a non-issue.

Reply:
Okay, will switch this over to use `GetSystemMetrics` instead :+1: